### PR TITLE
Allow for identifying when download failed due to violating a Downloa…

### PR DESCRIPTION
**Pull request by Konrad-Morawski**
…dBatchRequirementRule (as well as the specific rule that caused the error)

### CODE REVIEW

**Comment by hal9002** [comment]
Can one of the admins review this PR?

File: demo-simple/src/main/java/com/novoda/downloadmanager/demo/DownloadBatchSizeRequirementRule.java [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 public class DownloadBatchSizeRequirementRule implements DownloadBatchRequirementRule {
 
+    public final static int CODE = DownloadBatchRequirementRule.class.hashCode();
```
Comment (Konrad-Morawski) [inline demo-simple/src/main/java/com/novoda/downloadmanager/demo/DownloadBatchSizeRequirementRule.java:8]: For example (it's just a demo)

File: library/src/main/java/com/novoda/downloadmanager/DownloadError.java [hunk 2]
```diff
@@ -33,35 +45,37 @@ public String message() {
         return message;
     }
 
+    @Nullable
+    public Integer violatedRuleCode() {
```
Comment (Konrad-Morawski) [inline library/src/main/java/com/novoda/downloadmanager/DownloadError.java:49]: At the end of the chain, we'd retrieve the info (about a rule violation) from here

**Comment by Konrad-Morawski** [comment]
A false start on my part, disregard it for now please. Sorry

CC @hal9002

**Pull request closed without merging**